src/platform_impl/linux/x11/event_processor.rs

```diff
@@ -1503,6 +333,1225 @@ impl EventProcessor {
         }
     }
 
+    /// Get the platform window target.
+    pub fn window_target_mut(window_target: &mut RootELW) -> &mut EventLoopWindowTarget {
+        match &mut window_target.p {
+            PlatformEventLoopWindowTarget::X(target) => target,
+            #[cfg(wayland_platform)]
+            _ => unreachable!(),
+        }
+    }
+
+    fn client_message<T: 'static, F>(&mut self, xev: &XClientMessageEvent, mut callback: F)
+    where
+        F: FnMut(&RootELW, Event<T>),
+    {
+        let wt = Self::window_target(&self.target);
+        let atoms = wt.xconn.atoms();
+
+        let window = xev.window as xproto::Window;
+        let window_id = mkwid(window);
+
+        if xev.data.get_long(0) as xproto::Atom == wt.wm_delete_window {
+            let event = Event::WindowEvent {
+                window_id,
+                event: WindowEvent::CloseRequested,
+            };
+            callback(&self.target, event);
+            return;
+        }
+
+        if xev.data.get_long(0) as xproto::Atom == wt.net_wm_ping {
```

Comment:
Nit: We should use `else if` instead of `if`.

Reply:
it just was insanely long else if. in general i'd probably rewrite this to something else, but not right now.